File: Number_of_Laser_Beams_in_a_Bank.py
class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        res = prev = 0

        for r in range(len(bank)):
            cur = 0
            for c in range(len(bank[0])):
                if bank[r][c] == "1":
                    cur += 1
            
            res += cur * prev

            if cur > 0:
                prev = cur
        
        return res

File: Number_of_Laser_Beams_in_a_Bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int res = 0;
        int prev = 0;

        for (int r = 0; r < bank.size(); r++) {
            int cur = 0;
            for (int c = 0; c < bank[0].length(); c++) {
                if (bank[r][c] == '1') {
                    cur += 1;
                }
            }

            res += cur * prev;

            if (cur > 0) {
                prev = cur;
            }
        }

        return res;        
    }
};
